### ISSUE TITLE
Train/val split not randomized across training seeds

### ISSUE THREAD

**Issue by hyang0129** [hyang0129/HalluLens#34]
## Problem

AUROC std across seeds is suspiciously low (~0.0008), indicating the dataset is not actually randomized per seed.

### Root cause

In `scripts/run_experiment.py`, the `ActivationParser` is created **once** before the seed loop with a fixed `random_seed=split_seed` (default 42):

```python
ap = ActivationParser(
    ...,
    random_seed=experiment_cfg.get("split_seed", 42),  # always 42
    ...
)

for seed in seeds:
    seed_everything(seed)  # only affects weight init + global RNG
    # but AP's train/val split is already locked in
```

The train/val split happens in `ActivationParser.__init__` using `random_state=self.random_seed`, which never changes across seeds. So every seed trains and evaluates on the **exact same data split**, and the only variance comes from weight initialization.

### Fix

Add a `resplit(random_seed)` method to `ActivationParser` that re-applies `train_test_split` with a new seed, and call it at the start of each training seed iteration in `run_experiment.py`.

This ensures different seeds see different train/val splits, giving meaningful confidence intervals.
